WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteSourceManager.kt

```diff
@@ -0,0 +1,123 @@
+import org.wordpress.android.ui.mysite.cards.quickstart.QuickStartCardSource
+import org.wordpress.android.ui.mysite.dynamiccards.DynamicCardMenuViewModel.DynamicCardMenuInteraction
+import org.wordpress.android.ui.mysite.dynamiccards.DynamicCardMenuViewModel.DynamicCardMenuInteraction.Hide
+import org.wordpress.android.ui.mysite.dynamiccards.DynamicCardMenuViewModel.DynamicCardMenuInteraction.Pin
+import org.wordpress.android.ui.mysite.dynamiccards.DynamicCardMenuViewModel.DynamicCardMenuInteraction.Unpin
+import org.wordpress.android.ui.mysite.dynamiccards.DynamicCardsSource
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+import org.wordpress.android.util.config.MySiteDashboardPhase2FeatureConfig
+import javax.inject.Inject
+
+@Suppress("LongParameterList")
+class MySiteSourceManager @Inject constructor(
+    private val analyticsTrackerWrapper: AnalyticsTrackerWrapper,
+    private val currentAvatarSource: CurrentAvatarSource,
+    private val domainRegistrationSource: DomainRegistrationSource,
+    private val dynamicCardsSource: DynamicCardsSource,
+    private val quickStartCardSource: QuickStartCardSource,
+    private val scanAndBackupSource: ScanAndBackupSource,
+    private val selectedSiteSource: SelectedSiteSource,
+    postCardsSource: PostCardsSource,
+    siteIconProgressSource: SiteIconProgressSource,
+    private val mySiteDashboardPhase2FeatureConfig: MySiteDashboardPhase2FeatureConfig
+) {
+    private val mySiteSources: List<MySiteSource<*>> = listOf(
+            selectedSiteSource,
+            siteIconProgressSource,
+            quickStartCardSource,
+            currentAvatarSource,
+            domainRegistrationSource,
+            scanAndBackupSource,
+            dynamicCardsSource,
+            postCardsSource
+    )
+
+    fun build(siteLocalId: Int?, coroutineScope: CoroutineScope): List<LiveData<out PartialState>> {
+        return if (siteLocalId != null) {
+            mySiteSources.map { source -> source.build(coroutineScope, siteLocalId).distinctUntilChanged() }
+        } else {
+            mySiteSources.filterIsInstance(SiteIndependentSource::class.java)
+                    .map { source -> source.build(coroutineScope).distinctUntilChanged() }
+        }
+    }
+
+    fun isRefreshing(): Boolean {
+        if (mySiteDashboardPhase2FeatureConfig.isEnabled()) {
+            mySiteSources.filterIsInstance(MySiteRefreshSource::class.java).forEach {
+                if (it.isRefreshing() == true) {
+                    return true
+                }
+            }
+        }
+        return false
+    }
+
+    fun refresh() {
+        if (mySiteDashboardPhase2FeatureConfig.isEnabled()) {
+            refreshAllSources()
+        } else {
+            refreshSelectSources()
+        }
+    }
+
+    fun onResume(isFirstResume: Boolean) {
+        when (isFirstResume) {
+            true -> refreshSelectSources()
+            false -> if (mySiteDashboardPhase2FeatureConfig.isEnabled()) {
```

Comment:
Suggestion (💡): I now noticed that this `false` case can be further improved and become as simple as `false -> refresh()`. Interesting right, so much complexity is gone with step-by-step refactoring. 🎉

Reply:
Nice find!!!